Unicode UI/GDILibraryW.cls

```diff
@@ -0,0 +1,319 @@
+"Filed out from Dolphin Smalltalk 7"!
+
+GDILibrary subclass: #GDILibraryW
+	instanceVariableNames: ''
+	classVariableNames: ''
+	poolDictionaries: ''
+	classInstanceVariableNames: ''!
+GDILibraryW guid: (GUID fromString: '{80e1b07e-46ed-40eb-bf92-98d4dd955556}')!
+GDILibraryW comment: ''!
+!GDILibraryW categoriesForClass!External-Data-Structured-Win32! !
+!GDILibraryW methodsFor!
+
+addFontResource: fontFile
+
+^self addFontResourceW: fontFile!
+
+addFontResourceW: fontFile 
+	"Adds the font resource from the specified file to the system font table.
+	The font can subsequently be used for text output by any application. 
+
+	int AddFontResource(
+		LPCTSTR lpszFilename   // font file name
+	);"
+
+	<stdcall: sdword AddFontResourceW lpstr>
+	^self invalidCall!
+
+copyEnhMetaFile: hemf lpszFile: lpszFile
+
+^self copyEnhMetaFileW: hemf lpszFile: lpszFile!
+
+copyEnhMetaFileW: hemf lpszFile: lpszFile
+	"Invoke the CopyEnhMetaFile() function of the module wrapped by the receiver.
+	Helpstring: Copies contents of enhanced-format metafile to specified file.
+
+		HENHMETAFILE __stdcall CopyEnhMetaFile(
+			HENHMETAFILE hemf,
+			LPCSTR lpszFile);"
+
+	<stdcall: handle CopyEnhMetaFileW handle lpstr>
```

Comment:
There is an lpwstr type recognised by the compiler which has a separate type encoding. At present the VM does nothing special with this, but the intent was to allow it to perform auto-conversions to wide strings when passed byte strings. I think if this is done then in many cases (excluding APIs expecting or returning structs that contain strings) we can just switch over to always using the W version of the API.  I believe that much (if not most) of the time the A variant is just a little shim that does the transform to wide before calling the W version anyway, because the OS is natively using the wide strings. Consequently there should be major difference in perf.